include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h

```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
-	   *
-	   **/
-	  virtual void initialize(){};
-
-
 };
 
 /*************************************************************************
- * implementation of memeber execute()
+ * implementation of memebers
  * ***********************************************************************/
 
+/**
+ * @param ing reference to the object holding all information of the system
+ * @param filename output file name. defaults to "Rg2TimeSeries.dat".
+ * */
 template<class IngredientsType>
-AnalyzerRadiusOfGyration<IngredientsType>::AnalyzerRadiusOfGyration(const IngredientsType& ing,
-								    std::string filenamePrefix,
-								    uint64_t lowerMcsLimit)
-:ingredients(ing)
-,Rg_squaredX(1)
-,Rg_squaredY(1)
-,Rg_squaredZ(1)
-,Rg_squared(1)
-,lowerLimit(lowerMcsLimit)
-,outputFilePrefix(filenamePrefix)
-{
-	groups.push_back(MonomerGroup<molecules_type>((ing.getMolecules())));
-	
-	//add all monomers to the group
-	for(size_t n=0; n<ing.getMolecules().size();++n)
-	{
-		groups[0].push_back(n);
-	}
-}
-
-template<class IngredientsType>
-AnalyzerRadiusOfGyration<IngredientsType>::AnalyzerRadiusOfGyration(const IngredientsType& ing, 
-								    const MonomerGroup< molecules_type >& g
-								    ,std::string filenamePrefix
-								    ,uint64_t lowerMcsLimit)
+AnalyzerRadiusOfGyration<IngredientsType>::AnalyzerRadiusOfGyration(
+	const IngredientsType& ing, 
+	std::string filename)
 :ingredients(ing)
-,Rg_squaredX(1)
-,Rg_squaredY(1)
-,Rg_squaredZ(1)
-,Rg_squared(1)
-,lowerLimit(lowerMcsLimit)
-,outputFilePrefix(filenamePrefix)
+,bufferSize(100)
+,outputFile(filename)
+,isFirstFileDump(true)
 {
-	//add the group given as argument to groups. groups will only consist of this single group
-	groups.push_back(g);
 }
 
 
-template<class IngredientsType>
-AnalyzerRadiusOfGyration<IngredientsType>::AnalyzerRadiusOfGyration(const IngredientsType& ing, 
-								    const std::vector< MonomerGroup< molecules_type > >& g,
-								    std::string filenamePrefix,
-								    uint64_t lowerMcsLimit)
-:ingredients(ing)
-,groups(g)
-,Rg_squaredX(g.size())
-,Rg_squaredY(g.size())
-,Rg_squaredZ(g.size())
-,Rg_squared(g.size())
-,lowerLimit(lowerMcsLimit)
-,outputFilePrefix(filenamePrefix)
+/**
+ * @details fills all monomers into the groups vector as a single group,
+ * if the group size is zero. Normally this should always be the case,
+ * unless the groups were already set explicitly by using the setter function
+ * setMonomerGroups
+ * */
+template< class IngredientsType >
+void AnalyzerRadiusOfGyration<IngredientsType>::initialize()
 {
+	//if no groups are set, use the complete system by default
+	//groups can be set using the provided access function
+	if(groups.size()==0){
+		groups.push_back(MonomerGroup<molecules_type>(ingredients.getMolecules()));
+		for(size_t n=0;n<ingredients.getMolecules().size();n++)
+			groups[0].push_back(n);
+	}
+	//initialize the buffer 
+	Rg2TimeSeries.resize(4,std::vector<double>(0));
```

Comment:
The re-size of `Rg2TimeSeries.resize(4,std::vector<double>(0));` could be done in the initialization list in the constructor like this:
`..., Rg2TimeSeries(4,std::vector<double>()), ... `
This way, the vector is constructed with the proper length and is ready to use before calling initialize. And one does not need a extra call for this :o)